composer.json

```diff
@@ -53,7 +53,7 @@
     "php": "~7.3 || ~8",
     "composer-runtime-api": "~2.0",
     "dompdf/dompdf" : "~2.0.2",
-    "firebase/php-jwt": ">=3 <6",
+    "firebase/php-jwt": ">=3 <7",
```

Comment:
@seamuslee001 Ah, cool that this change looks as small as it does. 

The only thing is... I don't think `Firebase\JWT\Key` exists in all versions `>=3 <7`. (Skimming https://github.com/firebase/php-jwt/tree/v5.0.0/src, it looks like it was added somewhere during 5.x. But this constraint does reflect something else important - that the version-constraints are externally-influenced. (Just as we can't keep everyone on 5.x for as long as we please... we probably can't get everyone to 6.x when we please.)

In theory, if we were strictly supporting 6.x, then I think the loops/arrays could be simplified a lot. But there is another possibility -- take the approach you have above (2 changes; circa lines 76 and 82) and make those changes conditional:

```php
$useKeyObj = version_compare(\Composer\InstalledVersions::getVersion('firebase/php-jwt'), '6', '>=');
if (!$useKeyObj) Civi::log()->info("Using deprecated version of firebase/php-jwt. Upgrade to 6.x+.");
```
```php
$keysByAlg[$alg][$key['id']] = ($useKeyObj) 
  ? new Key($key['key'], $alg) 
  : $key['key'];
```
```php
return $useKeyObj 
  ?  (array) JWT::decode($token, $keys)
  : (array) JWT::decode($token, $keys, [$alg]);
```

Reply:
@totten ok I have added in that compatibility layer now